oauth2l-go/src/oauth2client/oauth2client.go

```diff
@@ -0,0 +1,313 @@
+package oauth2client
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	// The URN for getting verification token offline
+	oobCallbackUrn = "urn:ietf:wg:oauth:2.0:oob"
+	// The URN for token request grant type jwt-bearer
+	jwtBearerUrn = "urn:ietf:params:oauth:grant-type:jwt-bearer"
+)
+
+// Signer interface to support both rsa and ecdsa sign.
+type signerInterface interface {
+	sign(rand io.Reader, msg []byte, opts crypto.SignerOpts) ([]byte, error)
+}
+type rsaSigner struct {
+	pkey *rsa.PrivateKey
+}
+
+func (s rsaSigner) sign(rand io.Reader, msg []byte, opts crypto.SignerOpts) ([]byte, error) {
+	return s.pkey.Sign(rand, msg, opts)
+}
+
+type ecdsaSigner struct {
+	pkey *ecdsa.PrivateKey
+}
+
+func (s ecdsaSigner) sign(rand io.Reader, msg []byte, opts crypto.SignerOpts) ([]byte, error) {
+	return s.pkey.Sign(rand, msg, opts)
+}
+
+// Default function to handle authorize flow. This hanlder prints the URL on
```

Comment:
handle 3LO authorize flow. prints the authorization URL on stdout and reads the verification code form stdin.

Remove the (i.e. ...) part.